db/view/view_update_generator.cc

```diff
@@ -73,7 +133,7 @@ future<> view_update_generator::start() {
                                 tracing::trace_state_ptr ts,
                                 streamed_mutation::forwarding fwd_ms,
                                 mutation_reader::forwarding fwd_mr) {
-                        return ssts->make_range_sstable_reader(s, std::move(permit), pr, ps, pc, std::move(ts), fwd_ms, fwd_mr);
+                        return ssts->make_range_sstable_reader(s, std::move(permit), pr, ps, pc, std::move(ts), fwd_ms, fwd_mr, *_progress_tracker);
```

Comment:
progress_tracker is initialized to a null pointer. How are we sure that at this point it is not?

Reply:
Actually, we initialize progress tracker on ctor, and only set it to null on deferred action which only runs when the main loop exits. So progress_tracker cannot be null there.